src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +57,366 @@ uint32_t SystemRequest::index = 0;
+                QUERY_APPS_VALIDATION_FAILED
+                    << "An urlscheme length exceeds maximum allowed ["
+                    << app_data[os_type][json::urlScheme].asString().length()
+                    << "]>[255]");
+          }
+        }
+      } else if (app_data.keyExists(json::android)) {
+        os_type = json::android;
+        if (!app_data[os_type].keyExists(json::packageName)) {
+          LOG4CXX_WARN(logger_,
+                       QUERY_APPS_VALIDATION_FAILED
+                           << "Can't find package name in json file.");
+          result = false;
+        } else {
+          if (app_data[json::android][json::packageName].asString().length() >
+              255) {
+            LOG4CXX_WARN(logger_,
+                         QUERY_APPS_VALIDATION_FAILED
+                             << "Package name length ["
+                             << app_data[json::android][json::packageName]
+                                    .asString()
+                                    .length()
+                             << "] exceeds max lengthCan't find package name "
+                                "in json file.");
+            result = false;
+          }
+        }
+      }
+
+      if (os_type.empty()) {
+        LOG4CXX_WARN(logger_,
+                     QUERY_APPS_VALIDATION_FAILED
+                         << "Can't find mobile OS type in json file.");
+        result = false;
+        continue;
+      }
+
+      // languages verification
+      if (!app_data[os_type].keyExists(json::languages)) {
+        LOG4CXX_WARN(logger_, "\"languages\" not exists");
+        result = false;
+        continue;
+      }
+      if (!validateLanduages(app_data[os_type][json::languages]))
+        result = false;
+    }
+    return result;
+  }
+
+  bool validateAppIdAndAppName(
+      const smart_objects::SmartObject& app_data) const {
+    // verify appid
+    if (!app_data.keyExists(json::appId)) {
+      LOG4CXX_WARN(logger_,
+                   QUERY_APPS_VALIDATION_FAILED
+                       << "Can't find app ID in json file.");
+      return false;
+    }
+
+    // verify appid lenght
+    const std::string app_id(app_data[json::appId].asString());
+    if (app_id.length() > 100) {
+      LOG4CXX_WARN(logger_,
+                   QUERY_APPS_VALIDATION_FAILED
+                       << "An Object ID length exceeds maximum allowed ["
+                       << app_id.length() << "]>[100]");
+      return false;
+    }
+
+    // verify that appid is unique
+    if (app_ids_map_.find(app_id) != app_ids_map_.end()) {
+      LOG4CXX_WARN(logger_,
+                   QUERY_APPS_VALIDATION_FAILED
+                       << "An Object ID is not unigue [" << app_id << "]");
+      return false;
+    } else {
+      app_ids_map_[app_id] = 1;
+    }
+
+    // verify that app is not registered yet
+    ApplicationSharedPtr registered_app =
+        ApplicationManagerImpl::instance()->application_by_policy_id(app_id);
```

Comment:
Please try to avoid singleton Usage. 
You can pass ApplicationManagerImpl& as parameter to validators constructor and hold it.